indexing-service/src/main/java/org/apache/druid/indexing/overlord/TaskLockbox.java

```diff
@@ -197,17 +205,26 @@ public int compare(Pair<Task, TaskLock> left, Pair<Task, TaskLock> right)
           activeTasks.size(),
           storedLocks.size() - taskLockCount
       );
+
+      for (Task task : failedToReacquireLockTasks) {
+        for (TaskLock lock : taskStorage.getLocks(task.getId())) {
+          taskStorage.removeLock(task.getId(), lock);
+        }
+        taskStorage.setStatus(TaskStatus.failure(task.getId(), "Failed to reacquire lock"));
```

Comment:
This doesn't seem the right thing to do in `TaskLockbox`. If you notice, the `taskMaster` is used in this class to manipulate the locks and that's it. Lockbox doesn't modify any state other than the locks. We could maybe return the list of tasks for which we could not acquire locks to the caller. As you do this, it will be better to return a `Result` object where we can add more info later if needed. what do you think?

Reply:
done